controllers/webspherelibertyapplication_controller.go

```diff
@@ -103,6 +103,49 @@ func (r *ReconcileWebSphereLiberty) Reconcile(ctx context.Context, request ctrl.
 		ns = r.watchNamespaces[0]
 	}
 
+	// Fetch the WebSphereLiberty instance
+	instance := &webspherelibertyv1.WebSphereLibertyApplication{}
+	var ba common.BaseComponent = instance
+	err = r.GetClient().Get(context.TODO(), request.NamespacedName, instance)
+	if err != nil {
+		if kerrors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, err
+	}
+
+	defaultMeta := metav1.ObjectMeta{
+		Name:      instance.Name,
+		Namespace: instance.Namespace,
+	}
+
+	// Check to see if the existing deployment or statefulsets are owned by us
+	// If they are not, display and error and do not requeue
+	deploy := &appsv1.Deployment{ObjectMeta: defaultMeta}
+	err = r.GetClient().Get(context.TODO(), request.NamespacedName, deploy)
+	if err == nil {
+		for _, element := range deploy.OwnerReferences {
+			if element.Kind != "WebSphereLibertyApplication" {
+				reqLogger.Error(err, "Existing Deployment for "+instance.Name+" is not managed by this operator. It is being managed by "+element.Kind)
+				return reconcile.Result{}, nil
```

Comment:
I think we should report error back to the WLO CR. Otherwise new app will have no Reconciled status.
Besides Deployment and Statefulset ther conflict might occur with Knative Service

Reply:
Ok, I changed the code to update the status on the new app and tested that its working.

Reply:
I am not exactly sure what you meant by "Besides Deployment and Statefulset ther conflict might occur with Knative Service".  Let me know if this update doesn't resolve that concern as well.

Reply:
@smcclem We have an intergation with Knative. When it's enabled, neither Deployment or StatefulSet is created, instead Knative Service is created ([here](https://github.com/WASdev/websphere-liberty-operator/blob/147517030ab44f2c1b61bbef7005504d589e00d2/controllers/webspherelibertyapplication_controller.go#L327)), so we should check for that as well. Need to make sure that the Knative APIs are present on the cluster before checking the actual resource. Thank you.